### ISSUE TITLE
Permission denied error when shlink tries to download same GeoLite db more than once

### ISSUE THREAD

**Issue by schoeppe** [shlinkio/shlink#899]
<!--
Before opening an issue, just take into account that this is a completely free of charge and open source project.
I'm always happy to help and provide support, but some understanding will be expected.
I do this in my own free time, so expect some delays when implementing new features and fixing bugs, and don't take it personal if an issue gets eventually closed.
You may also be asked to provide tests or ways to reproduce reported bugs.
Try to be polite, and understand it is impossible for an OSS project to cover all use cases.

With that said, please fill in the information requested next. More information might be requested next (like logs or system configs).
-->

#### How Shlink is set-up

* Shlink Version: 2.4.0
* PHP Version: 7.4.12
* How do you serve Shlink: Self-hosted Apache
* Database engine used: MySQL

#### Summary

We have setup multiple installations on one server each served by a different vhost.
For geolocation we have configured cronjobs which run in the context of the user/group of the corresponding vhost.

#### Current behavior

When a Geolite2 database download is available and needs to be downloaded there is a conflict extracting the file Geolite2-City.mmdb file into the temporary directory. The first run of the task ./cli visit:locate creates a directory in the configured temp_dir "GeoLite2-City_[timestamp]" owned by the corresponding user. The second run of the task, started by another user, cannot extract the file to the already existing directory because of the directory ownership.

#### Expected behavior

I would expect multiple runs of the task visit:locate to be independent of each other, not conflicting in the described way.

#### How to reproduce

Just run the task ./cli visit:locate for two different installations with two different users.
Make sure there is no GeoLite2-City.mmdb file in the corresponding data-directory.

The first run succeeds, the second run fails with:
```Downloading GeoLite2 database...
 31546360/31546360 [============================] 100%PHP Warning:  PharData::extractTo(/tmp/GeoLite2-City_20201110/GeoLite2-City.mmdb): failed to open stream: Permission denied in /home/user2/htdocs/vendor/shlinkio/shlink-ip-geolocation/src/GeoLite2/DbUpdater.php on line 67


                                                                                                                        
 [ERROR] GeoLite2 database download failed. It is not possible to locate visits.                                        
                                                                                                                        

                                                                                                                        
 [ERROR] An error occurred while updating geolocation database, and an older version could not be found```

**Comment by schoeppe** [comment]
Related to https://github.com/shlinkio/shlink/issues/621

**Comment by acelaya** [comment]
Thanks for reporting.

I did that command in a way that it sets up a lock that prevents multiple executions in parallel, but I never considered having multiple instances of shlink where each one of them will have it's own lock, but a shared filesystem.

I think this should be easy to fix, by either using a more randomized temp name, or by using some folder inside shlink's instance dir.

Any is those approaches would prevent this collision.

**Comment by schoeppe** [comment]
Thanks for the quick reply!

To use the instance directory is a good idea. I updated the config in /home/user2/htdocs/config/params/generated_config.php and added:
```php
 'geolite2' => array(
        'license_key' => 'somelicense',
        'temp_dir' => '/home/user2/htdocs/data',
    ),
```
This solves the shared filesystem problem.
Is this the place to go for this configuration value?

**Comment by acelaya** [comment]
Yep. I actually didn't remember that config option even existed, haha.

It's probably as easy as that.

**Comment by schoeppe** [comment]
Ok, good. For me this issue is solved with the configuration change. Thank you!

> Yep. I actually didn't remember that config option even existed, haha.

I had a look at the repos and the code you implemented for this project... quite impressive.
The code is really well structured and easy to read - no need to keep every option in mind! :wink:
Good work! :+1:

**Comment by acelaya** [comment]
Thanks for the kind words :slightly_smiling_face:

**Comment by acelaya** [comment]
I have just merged a PR changing the default value for `temp_dir` so that it points to the `data` folder of the shlink instance itself, which should solve this problem.

The fix will be part of the upcoming v2.4.2
